### PR TITLE
Verify SSL in grafana and mimir

### PR DESCRIPTION
**Pull request by jayeshsadhwani99**

### CODE REVIEW

File: connectors/models.py [hunk 1]
```diff
@@ -333,7 +336,6 @@ def proto(self):
                              SourceKeyType.GCM_PROJECT_ID, SourceKeyType.GCM_PRIVATE_KEY,
                              SourceKeyType.GCM_CLIENT_EMAIL, SourceKeyType.PAGER_DUTY_API_KEY,
                              SourceKeyType.POSTGRES_PASSWORD, SourceKeyType.POSTGRES_USER,
-                             SourceKeyType.GRAFANA_API_KEY,
```
Comment (dimittal) [inline connectors/models.py:336]: Why is this removed?

Comment (jayeshsadhwani99) [reply connectors/models.py:336]: because it was duplicate

**Review by dimittal — DISMISSED** [review]

File: connectors/crud/connectors_crud.py [hunk 2]
```diff
@@ -90,12 +90,16 @@ def generate_credentials_dict(connector_type, connector_keys):
...
                 credentials_dict['grafana_host'] = conn_key.key.value
+            elif conn_key.key_type == SourceKeyType.SSL_VERIFY:
+                credentials_dict['ssl_verify'] = conn_key.key.value
     elif connector_type == Source.GRAFANA_VPC:
         for conn_key in connector_keys:
             if conn_key.key_type == SourceKeyType.AGENT_PROXY_API_KEY:
                 credentials_dict['agent_proxy_api_key'] = conn_key.key.value
             elif conn_key.key_type == SourceKeyType.AGENT_PROXY_HOST:
                 credentials_dict['agent_proxy_host'] = conn_key.key.value
+            elif conn_key.key_type == SourceKeyType.SSL_VERIFY:
```
Comment (droid-mohit) [inline connectors/crud/connectors_crud.py:101]: Remove it from here

File: connectors/models.py [hunk 3]
```diff
@@ -95,13 +95,14 @@
     Source.GRAFANA: [
         [
             SourceKeyType.GRAFANA_HOST,
-            SourceKeyType.GRAFANA_API_KEY
+            SourceKeyType.GRAFANA_API_KEY,
+            SourceKeyType.SSL_VERIFY,
```
Comment (droid-mohit) [inline connectors/models.py:99]: Create another entry with just this combination: 
[
            SourceKeyType.GRAFANA_HOST,
            SourceKeyType.GRAFANA_API_KEY
]

File: connectors/models.py [hunk 4]
```diff
@@ -95,13 +95,14 @@
...
         ]
     ],
     Source.GRAFANA_VPC: [
         [
             SourceKeyType.AGENT_PROXY_HOST,
-            SourceKeyType.AGENT_PROXY_API_KEY
+            SourceKeyType.AGENT_PROXY_API_KEY,
```
Comment (droid-mohit) [inline connectors/models.py:105]: No need to add ',' here

File: connectors/models.py [hunk 5]
```diff
@@ -157,6 +158,7 @@
         [
             SourceKeyType.MIMIR_HOST,
             SourceKeyType.X_SCOPE_ORG_ID,
+            SourceKeyType.SSL_VERIFY,
```
Comment (droid-mohit) [inline connectors/models.py:161]: Remove it from here

File: connectors/models.py (hunk 3 above)
Comment (droid-mohit) [reply connectors/models.py:99]: You havent added an entry for GRAFANA_MIMIR?

File: executor/metric_task_executor/mimir_task_executor.py [hunk 6]
```diff
@@ -44,6 +44,8 @@ def __init__(self, account_id):
                 self.__mimir_host = key.key
             if key.key_type == SourceKeyType.X_SCOPE_ORG_ID:
                 self.__x_scope_org_id = key.key
+            elif key.key_type == SourceKeyType.SSL_VERIFY:
+                self.__ssl_verify = key.key
```
Comment (droid-mohit) [inline executor/metric_task_executor/mimir_task_executor.py:48]: set the default value to True
Add 
self.__ssl_verify = True on line number 41

File: executor/metric_task_executor/grafana_executor.py [hunk 7]
```diff
@@ -43,6 +43,8 @@ def __init__(self, account_id):
                 self.__grafana_api_key = key.key
             elif key.key_type == SourceKeyType.GRAFANA_HOST:
                 self.__grafana_host = key.key
+            elif key.key_type == SourceKeyType.SSL_VERIFY:
+                self.__ssl_verify = key.key
```
Comment (droid-mohit) [inline executor/metric_task_executor/grafana_executor.py:47]: set the default value to True
Add
self.__ssl_verify = True on line number 40

File: integrations_api_processors/grafana_api_processor.py [hunk 8]
```diff
@@ -8,17 +8,19 @@
 class GrafanaApiProcessor:
     client = None
 
-    def __init__(self, grafana_host, grafana_api_key):
+    def __init__(self, grafana_host, grafana_api_key, ssl_verify="true"):
```
Comment (droid-mohit) [inline integrations_api_processors/grafana_api_processor.py:11]: This value should be in bool and not string

File: integrations_api_processors/grafana_api_processor.py [hunk 9]
```diff
@@ -8,17 +8,19 @@
...
+        self.__ssl_verify = ssl_verify
         self.headers = {
             'Authorization': f'Bearer {self.__api_key}'
         }
 
     def test_connection(self):
         try:
             url = '{}/api/datasources'.format(self.__host)
-            response = requests.get(url, headers=self.headers)
+            verify = self.__ssl_verify == "true"
```
Comment (droid-mohit) [inline integrations_api_processors/grafana_api_processor.py:22]: Dont convert it here, make the conversion in parent class

File: integrations_api_processors/grafana_api_processor.py [hunk 10]
```diff
@@ -32,7 +34,8 @@ def test_connection(self):
     def fetch_data_sources(self):
         try:
             url = '{}/api/datasources'.format(self.__host)
-            response = requests.get(url, headers=self.headers)
+            verify = self.__ssl_verify == "true"
```
Comment (droid-mohit) [inline integrations_api_processors/grafana_api_processor.py:37]: Dont convert it here, make the conversion in parent class

File: integrations_api_processors/grafana_api_processor.py [hunk 11]
```diff
@@ -42,7 +45,8 @@ def fetch_data_sources(self):
     def fetch_dashboards(self):
         try:
             url = '{}/api/search'.format(self.__host)
-            response = requests.get(url, headers=self.headers)
+            verify = self.__ssl_verify == "true"
```
Comment (droid-mohit) [inline integrations_api_processors/grafana_api_processor.py:48]: Dont convert it here, make the conversion in parent class

File: integrations_api_processors/grafana_api_processor.py [hunk 12]
```diff
@@ -52,7 +56,8 @@ def fetch_dashboards(self):
     def fetch_dashboard_details(self, uid):
         try:
             url = '{}/api/dashboards/uid/{}'.format(self.__host, uid)
-            response = requests.get(url, headers=self.headers)
+            verify = self.__ssl_verify == "true"
```
Comment (droid-mohit) [inline integrations_api_processors/grafana_api_processor.py:59]: Dont convert it here, make the conversion in parent class

File: integrations_api_processors/grafana_api_processor.py [hunk 13]
```diff
@@ -64,7 +69,8 @@ def fetch_promql_metric_labels(self, promql_datasource_uid, metric_name):
         try:
             url = '{}/api/datasources/proxy/uid/{}/api/v1/labels?match[]={}'.format(self.__host, promql_datasource_uid,
                                                                                     metric_name)
-            response = requests.get(url, headers=self.headers)
+            verify = self.__ssl_verify == "true"
```
Comment (droid-mohit) [inline integrations_api_processors/grafana_api_processor.py:72]: Dont convert it here, make the conversion in parent class

File: integrations_api_processors/grafana_api_processor.py [hunk 14]
```diff
@@ -76,7 +82,8 @@ def fetch_promql_metric_label_values(self, promql_datasource_uid, metric_name, l
             url = '{}/api/datasources/proxy/uid/{}/api/v1/label/{}/values?match[]={}'.format(self.__host,
                                                                                              promql_datasource_uid,
                                                                                              label_name, metric_name)
-            response = requests.get(url, headers=self.headers)
+            verify = self.__ssl_verify == "true"
```
Comment (droid-mohit) [inline integrations_api_processors/grafana_api_processor.py:85]: Dont convert it here, make the conversion in parent class

File: integrations_api_processors/grafana_api_processor.py [hunk 15]
```diff
@@ -87,7 +94,8 @@ def fetch_promql_metric_timeseries(self, promql_datasource_uid, query, start, en
         try:
             url = '{}/api/datasources/proxy/uid/{}/api/v1/query_range?query={}&start={}&end={}&step={}'.format(
                 self.__host, promql_datasource_uid, query, start, end, step)
-            response = requests.get(url, headers=self.headers)
+            verify = self.__ssl_verify == "true"
```
Comment (droid-mohit) [inline integrations_api_processors/grafana_api_processor.py:97]: Dont convert it here, make the conversion in parent class

File: integrations_api_processors/mimir_api_processor.py [hunk 16]
```diff
@@ -7,14 +7,16 @@
 class MimirApiProcessor:
     client = None
 
-    def __init__(self, mimir_host, x_scope_org_id='anonymous'):
```
Comment (droid-mohit) [inline integrations_api_processors/mimir_api_processor.py:10]: Same as above

**Review by droid-mohit — APPROVED** [review]

**Review by droid-mohit — APPROVED** [review]
